eclipse-collections/src/main/java/org/eclipse/collections/impl/bag/mutable/UnmodifiableBag.java

```diff
@@ -438,4 +439,10 @@ protected Object readResolve()
     {
         return this.getMutableBag().selectUnique();
     }
+
+    @Override
+    public SetIterable<T> asSet()
+    {
+        return this.getMutableBag().asSet();
```

Comment:
@motlin I can't call `asUnmodifiable()` because the return type of `asSet()` here is `SetIterable`: I am expecting it to be unmodifiable if needed be. I will add a test to ensure it is the case though!

Reply:
@motlin actually the test was already there :)